### PR TITLE
Add multiarch

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Needed for devel on Mac MX

### CODE REVIEW

File: .github/workflows/images.yaml [hunk 1]
```diff
@@ -65,7 +67,8 @@ jobs:
           CONTROLLER_GEN_VERSION: "v0.9.2"
           CLIENT_GEN_VERSION: "v0.26.1"
         run: |
-          docker build --build-arg GOLANG_VERSION=${GOLANG_VERSION} \
+          docker build --platform=linux/amd64,linux/arm64 \
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply .github/workflows/images.yaml:70]: yeah, I am looking how to

[the remark this answers is not in the document]
Comment (elezar) [reply .github/workflows/images.yaml:70]: For example, for the device plugin:
```
DOCKER_BUILDKIT=1 \
        docker buildx build --pull \
        --output=type=image,push=false \
        --platform=linux/amd64,linux/arm64 \
        --tag nvcr.io/nvidia/k8s-device-plugin:v0.14.3-ubuntu20.04 \
        --build-arg BASE_DIST="ubuntu20.04" \
        --build-arg CUDA_VERSION="12.3.0" \
        --build-arg GOLANG_VERSION="1.20.5" \
        --build-arg VERSION="v0.14.3" \
        --build-arg GIT_COMMIT="2bef25804caf5924f35a164158f097f954fe4c74-dirty" \
         \
        -f /Users/elezar/src/device-plugin/deployments/container/Dockerfile.ubuntu \
        /Users/elezar/src/device-plugin
```

**Review by elezar — COMMENTED** [review]
Is there any value in separating the build and push jobs if the only different *should* be:
```
 push: false
```
vs
```
 push: true
```
(and possibly the architecture?)

Or could you better describe the interaction between the two jobs?

File: .github/workflows/push.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Build and Push
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: .
+          outputs: "type=registry,push=true"
+          file: ./images/devel/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            GOLANG_VERSION=${{ matrix.go-version }}
```
Comment (elezar) [inline .github/workflows/push.yaml:50]: Let's sort the `build-args`.

**Comment by ArangoGutierrez** [comment]
> Is there any value in separating the build and push jobs if the only different _should_ be:
> 
> ```
>  push: false
> ```
> 
> vs
> 
> ```
>  push: true
> ```
> 
> (and possibly the architecture?)
> 
> Or could you better describe the interaction between the two jobs?

The just build, only build major X.X go version and on the default platform, not to waste time on building the big matrix.
Once the PR is in, the PUSH action, will build on arm and amd64 and on all the required go versions x.y.z

**Comment by elezar** [comment]
OK. That's fair. My concern is drift between the two jobs. Could we construct the matrix conditionally as per https://stackoverflow.com/questions/65384420/how-do-i-make-a-github-action-matrix-element-conditional (as well as `push=true`) and reuse the same step definition?

**Review by elezar — APPROVED** [review]
I still feel that we should reevaluate reusing the steps more to not have to update these in multiple places, but I don't see this as a blocker for moving forward in other projects.

Feel free to merge this to unblock and revisit the refactor in a follow-up.

**Pull request merged**